controllers/datadogagent/feature/kubernetesstatecore/feature.go

```diff
@@ -59,47 +59,49 @@ type ksmFeature struct {
 // Configure use to configure the feature from a v2alpha1.DatadogAgent instance.
 func (f *ksmFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredComponents {
 	f.owner = dda
+	var output feature.RequiredComponents
+
 	if dda.Spec.Features.KubeStateMetricsCore != nil && apiutils.BoolValue(dda.Spec.Features.KubeStateMetricsCore.Enabled) {
 		f.enable = true
+		output.ClusterAgent.IsRequired = &f.enable
 
 		if dda.Spec.Features.KubeStateMetricsCore.Conf != nil {
 			f.customConfig = v2alpha1.ConvertCustomConfig(dda.Spec.Features.KubeStateMetricsCore.Conf)
 		}
 
 		f.configConfigMapName = apicommonv1.GetConfName(dda, f.customConfig, apicommon.DefaultKubeStateMetricsCoreConf)
-	}
 
-	if dda.Spec.Features.ClusterChecks != nil && apiutils.BoolValue(dda.Spec.Features.ClusterChecks.Enabled) {
-		f.clusterChecksEnabled = true
-		if apiutils.BoolValue(dda.Spec.Features.ClusterChecks.UseClusterChecksRunners) {
-			f.rbacSuffix = common.CheckRunnersSuffix
-			f.serviceAccountName = v2alpha1.GetClusterChecksRunnerServiceAccount(dda)
-		} else {
-			f.serviceAccountName = v2alpha1.GetClusterAgentServiceAccount(dda)
+		if dda.Spec.Features.ClusterChecks != nil && apiutils.BoolValue(dda.Spec.Features.ClusterChecks.Enabled) {
+			f.clusterChecksEnabled = true
+			output.ClusterCheckRunner.IsRequired = &f.clusterChecksEnabled
```

Comment:
should this be set to `true` under `if apiutils.BoolValue(dda.Spec.Features.ClusterChecks.UseClusterChecksRunners)` ?

Reply:
right fixing now

Reply:
@celenechang 
I made the fix in https://github.com/DataDog/datadog-operator/pull/496